### PR TITLE
#2284 LFRicRunTimeChecks Split from dynamo0p3.py

### LINKED ISSUE #2284: Splitting LFRicRunTimeChecks from src/psyclone/dynamo0p3.py
Sub-issue of #2235

### PR DESCRIPTION
**Pull request by oakleybrunt**
Closes #2284

Sub_issue of #2235

### CODE REVIEW

**Comment by oakleybrunt** [comment]
LFRicRunTimeChecks requires LFRicCollection.
LFRicCollection requires LFRicInvoke.
LFRicInvoke requires LFRicRunTimeChecks.

File: src/psyclone/domain/lfric/lfric_run_time_checks.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        :param parent: the node in the f2pygen AST representing the PSy- \
+                       layer routine.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        parent.add(CommentGen(
+            parent, " Check field function space and kernel metadata "
+            "function spaces are compatible"))
+
+        # When issue #753 is addressed (with issue #79 helping further)
```
Comment (arporter) [inline src/psyclone/domain/lfric/lfric_run_time_checks.py:93]: #753 has been closed so I suggest replacing it with #30.

File: src/psyclone/domain/lfric/lfric_run_time_checks.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        indirectly pick up a readonly field being modified, it would
+        not be picked up where the error occured. Therefore adding
+        checks here is still useful.
+
+        :param parent: the node in the f2pygen AST representing the PSy- \
+                       layer routine.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        # When issue #753 is addressed (with issue #79 helping further)
```
Comment (arporter) [inline src/psyclone/domain/lfric/lfric_run_time_checks.py:168]: Again, replace 753 with 30 please.

**Review by arporter — CHANGES_REQUESTED** [review]
Almost perfect! Just a couple of mentions of an issue that has been closed so the comments need updating.
Also, this has conflicts with master so I haven't been able to check the CI  or docs yet.

**Review by arporter — COMMENTED** [review]
Requested changes have been made.
Ref. guide builds fine.
CI is all green.
I'll fire off the integration tests.

**Review by arporter — APPROVED** [review]
Integration tests all green.

**Pull request merged**